src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -246,38 +248,91 @@ snc_quota_manager::calc_node_quota_default() const {
     return default_quota;
 }
 
+void snc_quota_manager::get_or_create_quota_context(
+  std::unique_ptr<snc_quota_context>& ctx,
+  std::optional<std::string_view> client_id) {
+    if (likely(ctx)) {
+        // NB: comparing string_view to sstring might be suboptimal
+        if (likely(ctx->_client_id == client_id)) {
+            // the context is the right one
+            return;
+        }
+
+        // either of the context indexing propeties have changed on the client
+        // within the same connection. This is an unexpected path, quotas may
+        // misbehave if we ever get here. The design is based on assumption that
+        // this should not happen. If it does happen with a supported client, we
+        // probably should start supporting multiple quota contexts per
+        // connection
+        vlog(
+          klog.warn,
+          "qm - client_id has changed on the connection. Quotas are reset now. "
+          "Old client_id: {}, new client_id: {}",
+          ctx->_client_id,
+          client_id);
+    }
+
+    ctx = std::make_unique<snc_quota_context>(client_id);
+    const auto tcgroup_it = config::find_throughput_control_group(
+      _kafka_throughput_control().cbegin(),
+      _kafka_throughput_control().cend(),
+      client_id);
+    if (tcgroup_it == _kafka_throughput_control().cend()) {
+        ctx->_exempt = false;
+        vlog(klog.info, "qm - No throghput control group assigned");
```

Comment:
should these be at `info` level?

Reply:
In the normal course that would happen once per connection lifetime, so I think info is fine. Unless we have supported clients that open and drop kafka connections all the time.

Reply:
do we log anything else at connection establishment at info level? i don't think we do...

Reply:
we log at INFO at connection loss [here](https://github.com/redpanda-data/redpanda/blob/dev/src/v/net/server.cc#L154-L159)